File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <string.h>

int main(void)
{
    int n, j = 0;
    string text = get_string("Text: ");
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] < 65 || text[i] > 122 || (text[i] < 97 && text[i] > 90))
        {
            j++;
        }
        n = i + 1 - j;
    }
    j = 0;
    printf("%i letter(s)", n);
    for (int i = 0; i <= strlen(text); i++)
    {
        if (text[i] == 32 && (text[i + 1] != 32 || text[i + 1] != 0))
        {
            j++;
        }
    }
    int w = j + 1;
    printf("\n%i word(s)", w);
    j = 0;
    for (int i = 0; i <= strlen(text); i++)
    {
        if (text[i] == 63 || text[i] == 33 || text[i] == 46)
        {
            j++;
        }
    }
    int s = j;
    printf("\n%i sentence(s)", s);
    float L = (float) n / w * 100;
    float S = (float) s / w * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int result = round(index);
    if (result < 1)
    {
        printf("\nBefore Grade 1\n");
    }
    else if (result >= 16)
    {
        printf("\nGrade 16+\n");
    }
    else
    {
        printf("\nGrade %i\n", result);
    }
}

File: readability.py
from cs50 import get_string

text = get_string("Text: ")
index = 0
count_letter = 0
count_word = 0
count_sentence = 0

for i in range(len(text)):
    if text[index].isalpha():
        count_letter += 1
    index += 1
    
index = 0

for i in range(len(text)):
    if text[index] == ' ':
        count_word += 1
    index += 1
count_word += 1

index = 0

for i in range(len(text)):
    if text[index] == '.' or text[index] == '?' or text[index] == '!':
        count_sentence += 1
    index += 1
    
L = count_letter / count_word * 100
S = count_sentence / count_word * 100

result = round(0.0588 * L - 0.296 * S - 15.8)

if result < 1:
    print("Before Grade 1")
elif result >= 16:
    print("Grade 16+")
else:
    print(f"Grade {result}")